crypto/merkle/hash.go

```diff
@@ -22,5 +22,9 @@ func leafHash(leaf []byte) []byte {
 
 // returns tmhash(0x01 || left || right)
 func innerHash(left []byte, right []byte) []byte {
-	return tmhash.Sum(append(innerPrefix, append(left, right...)...))
+	data := make([]byte, 0, len(innerPrefix)+len(left)+len(right))
+	data = append(data, innerPrefix...)
+	data = append(data, left...)
+	data = append(data, right...)
```

Comment:
How about using copy?
```go
n := copy(data, innerPrefix)
n += copy(data[n:], left)
n += copy(data[n:], right)
```
Please try that and run some benchmarks then let me know what's up.

Reply:
Awesome, let's use it then! I believe that copy translates into a MEMMOVE which is much much faster and as the bytes to be hashed increased we should even more radical improvements.

Reply:
Also please don't forget to use `tmhash.Sum(data[:n])` :-)